README.md

```diff
@@ -214,6 +214,28 @@ you print in code outside of `@test`, `setup` or `teardown` functions
 must be redirected to `stderr` (`>&2`). Otherwise, the output may
 cause Bats to fail by polluting the TAP stream on `stdout`.
 
+### File descriptor 3
+
+Bats makes a separation between output from the code under test and output that forms the TAP stream (which is produced by Bats internals). This is done in order to produce TAP-compliant output. In the [Printing to the terminal](#printing-to-the-terminal) section, there are details on how to use file descriptor 3 to print custom text properly. 
+
+A side-effect of using file descriptor 3 is that, under some circumstances, it can cause Bats to block and execution to seem dead without reason. This can happen if a child process is spawned in the background from a test. In this case, the child process will inherit file descriptor 3. Bats, as the parent process, will wait for the file descriptor to be closed by the child process before continuing execution. If the child process takes a lot of time to complete (eg if the child process is a `sleep 100` command or a background service that will run indefinitely), Bats will be similarly blocked for the same amount of time. To prevent this from happening, you need to close the file descriptor when calling the child process, eg `command_name 3>- &`.
+
+### Printing to the terminal
```

Comment:
I don't think we actually want to encourage this. If there is a legitimate reason for users to inject output into the TAP stream, we should provide concrete examples. Otherwise, we should remove this section.

Reply:
@mbland TBH, I included this information for selfish reasons. I had often spent time going through the same code over and over again to understand exactly what goes on inside Bats and why, that I decided to put it down to paper. I often find myself in need to debug my test scripts and printing from different places is crucial. But your comment made be curious and I looked into the TAP specification, version 12. It mentions for lines starting with `#`:

> Additional information may be put into the testing output on separate lines. Diagnostic lines should begin with a #, which the harness must ignore, at least as far as analyzing the test results. The harness is free, however, to display the diagnostics.

And for arbitrary lines not starting with `#`:

> Any output line that is not a version, a plan, a test line, a diagnostic or a bail out is considered an “unknown” line. A TAP parser is required to not consider an unknown line as an error but may optionally choose to capture said line and hand it to the test harness, which may have custom behavior attached. This is to allow for forward compatability. 

Version 13 agrees on custom diagnostics (although it seems to deprecated them in favor of yaml blocks) but considers random lines without `#` an error:

> Any output that is not a version, a plan, a test line, a YAML block, a diagnostic or a bail out is incorrect.

So, for custom diagnostics lines I think it is useful to have these instruction at least until sth like #64 is implemented. It is valid behavior mentioned in the spec that Bats is capable of exhibiting, but it is jot documented anywhere explicitly.

For non diagnostic, arbitrary lines, I agree that it is a bad idea to use them. They can be useful in debugging, but the same debugging can be with the "more" proper diagnostic lines. I think I read somewhere that Bats is version 12 compliant, in which case it is not wrong behavior, though.

But, apart from the above, I think this passage is also useful for understanding how Bats works internally. Maybe it is too much for a README and it makes more sense in a wiki page (but this also brings the issue of maintaining two complementary but separate documentations).